eslint.config.js

```diff
@@ -0,0 +1,64 @@
+const js = require('@eslint/js');
+const globals = require('globals');
+
+
+const styleRules = {
+    quotes: ['error', 'single', { avoidEscape: true }],
+    'no-trailing-spaces': 'error',
+    indent: 'error',
+    'linebreak-style': ['error', 'unix'],
+    semi: ['error', 'always'],
+    'brace-style': ['error', '1tbs', { allowSingleLine: true }],
+    'keyword-spacing': 'error',
+    'space-before-blocks': 'error',
+    'space-before-function-paren': [
+        'error',
+        { anonymous: 'always', named: 'never' },
+    ],
+    'no-mixed-spaces-and-tabs': 'error',
+    'comma-spacing': ['error', { before: false, after: true }],
+    'key-spacing': ['error', { beforeColon: false, afterColon: true }],
+};
+
+
+module.exports = [
+    js.configs.recommended,
+    {
+        languageOptions: {
+            ecmaVersion: 2022,
+            globals: {
+                ...globals.node,
+                ...globals.browser,
+                request: 'readonly',
+                response: 'readonly',
+                expect: 'readonly',
+                describe: 'readonly',
+                it: 'readonly',
+                render: 'readonly',
+                cleanHtml: 'readonly',
+                beforeEach: 'readonly',
+                sinon: 'readonly',
+                document: 'readonly'
+            }
+        },
+        rules: {
+            'no-unused-vars': [
+                'error',
+                { argsIgnorePattern: '^(err|req|res|next)$' },
+            ],
+            'one-var': ['error', { initialized: 'never' }],
+            'no-var': 'error',
+            ...styleRules,
+        },
+    },
```

Comment:
We can add another config object here targeting `components/**/*.js` (and possibly ignoring `components/**/spec.*.js` if required depending on how they play with the client-side JS test config) to override the the rules we don't want to apply to client-side JS e.g. `no-var`. The client-side JS config should also set `ecmaVersion: 2015` for maximum compatibility.

See our internal ESLint config for how it configures linting of client-side JS.